### PR TITLE
boardd: new class USBDevice

### PR DESCRIPTION
**Pull request by deanlee**

### CODE REVIEW

File: selfdrive/boardd/SConscript [hunk 1]
```diff
@@ -1,9 +1,9 @@
 Import('env', 'envCython', 'common', 'cereal', 'messaging')
 
 libs = ['usb-1.0', common, cereal, messaging, 'pthread', 'zmq', 'capnp', 'kj']
-env.Program('boardd', ['boardd.cc', 'panda.cc', 'pigeon.cc'], LIBS=libs)
+env.Program('boardd', ['boardd.cc', 'panda.cc',  'usbdevice.cc', 'pigeon.cc'], LIBS=libs)
```
Comment (adeebshihadeh) [inline selfdrive/boardd/SConscript:4]:
```suggestion
env.Program('boardd', ['boardd.cc', 'panda.cc', 'usbdevice.cc', 'pigeon.cc'], LIBS=libs)
```

File: selfdrive/boardd/usbdevice.h [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+#include <atomic>
+#include <mutex>
+#include <vector>
+
+#include <libusb-1.0/libusb.h>
+
+#define TIMEOUT 0
+#define PANDA_VENDOR_ID 0xBBAA
+#define PANDA_PRODUCT_ID 0xDDCC
```
Comment (adeebshihadeh) [inline selfdrive/boardd/usbdevice.h:11]: these should go in a header in panda

File: selfdrive/boardd/panda.h [hunk 3]
```diff
@@ -64,35 +56,18 @@ struct can_frame {
 	long src;
 };
 
-class Panda {
- private:
-  libusb_context *ctx = NULL;
-  libusb_device_handle *dev_handle = NULL;
-  std::mutex usb_lock;
+class Panda : public USBDevice{
```
Comment (adeebshihadeh) [inline selfdrive/boardd/panda.h:59]:
```suggestion
class Panda : public USBDevice {
```

File: selfdrive/boardd/panda.h [hunk 4]
```diff
@@ -64,35 +56,18 @@ struct can_frame {
...
+private:
```
Comment (adeebshihadeh) [inline selfdrive/boardd/panda.h:60]: move private to bottom?

File: selfdrive/boardd/usbdevice.h (hunk 2 above)
Comment (briskspirit) [reply selfdrive/boardd/usbdevice.h:11]: Will need to refactor panda headers slightly tomorrow, also will move other defines like RECV_SIZE etc.
Think bus count also should be included from panda?

Comment (adeebshihadeh) [reply selfdrive/boardd/usbdevice.h:11]: yeah that sounds good

**Comment by deanlee** [comment]
reopened ,it's become a new PR.

**Pull request merged**